metrics/otel_metrics.go

```diff
@@ -124,6 +135,46 @@ func (o *OTelMetrics) Start() error {
 		sdkmetric.WithResource(res),
 	)
 	o.meter = provider.Meter("otelmetrics")
+
+	// These metrics are dynamic fields that should always be collected
+	name := "num_goroutines"
+	var fgo metric.Float64Callback = func(_ context.Context, result metric.Float64Observer) error {
+		result.Observe(float64(runtime.NumGoroutine()))
+		return nil
+	}
+	g, err := o.meter.Float64ObservableGauge(name, metric.WithFloat64Callback(fgo))
+	if err != nil {
+		return err
+	}
+	o.gauges[name] = g
+
+	name = "memory_inuse"
+	// This is just reporting the gauge we already track under a different name.
+	var fmem metric.Float64Callback = func(_ context.Context, result metric.Float64Observer) error {
+		o.lock.RLock()
+		v, _ := o.Get("memory_heap_allocation")
```

Comment:
What is the value of `v` if this returns an error (I'm assuming `_` is an error)?  Do we need to capture that error?

Reply:
It's not an error, it's an "ok" value telling you that the value you queried was in the dataset. v will be 0 in that case, which should never occur because this is duplicating the other metric. Get is basically just a map query, but the ok value is not optional in that case. I mean, I could return a "can't happen" error here, but 0 is probably better.